src/support/migrate/providers/elasticsearch.ts

```diff
@@ -0,0 +1,96 @@
+import { Client } from "@elastic/elasticsearch";
+import { Provider } from ".";
+
+export type ElasticsearchProviderOptions = {
+  scrollDuration: string;
+  batchSize: number;
+};
+
+export class Elasticsearch implements Provider {
+  private readonly client: Client;
+  private readonly options: ElasticsearchProviderOptions;
+
+  constructor(url: string, options: ElasticsearchProviderOptions) {
+    this.client = new Client({ node: url });
+    this.options = options;
+  }
+
+  async listIndices(pattern?: string): Promise<string[]> {
+    const { body } = await this.client.cat.indices({ format: "json" });
+
+    return body
+      .map(({ index }: { index: string }) => index)
+      .filter((index: string) => index.match(pattern as string))
+      .filter((index: string) => !index.startsWith(".")) // ignore system indices
+      .sort();
+  }
+
+  async getIndex(index: string): Promise<any> {
+    const resp = await this.client.indices.get({ index });
+
+    // Remove unsupported settings properties
+    const specifications = resp.body[index as keyof typeof resp];
+    delete specifications.settings.index.creation_date;
+    delete specifications.settings.index.provided_name;
+    delete specifications.settings.index.uuid;
+    delete specifications.settings.index.version;
+    delete specifications.settings.index.routing;
+
+    return specifications;
+  }
+
+  async createIndex(index: string, specifications: any): Promise<void> {
+    specifications.settings.write = {
+      wait_for_active_shards: 'all'
+    };
+    await this.client.indices.create({ index, body: specifications });
+  }
+
+  async readDataFirst(index: string): Promise<any> {
+    const { body } = await this.client.search({
+      index,
+      scroll: this.options.scrollDuration,
```

Comment:
Instead of scroll query, Elastic says to use sliced point in time queries that can be run in parallel:
https://www.elastic.co/guide/en/elasticsearch/reference/current/point-in-time-api.html#search-slicing

Reply:
Good catch, I'm creating an issue to test and replace scroll by Point in Time usage

Reply:
https://github.com/kuzzleio/kourou/issues/172